src/IssueViz.Security/ObservingIssueLocationStore.cs

```diff
@@ -0,0 +1,123 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2020 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.ComponentModel.Composition;
+using System.Linq;
+using SonarLint.VisualStudio.Core.Helpers;
+using SonarLint.VisualStudio.IssueVisualization.Editor.LocationTagging;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security
+{
+    internal interface IObservingIssueLocationStore : IIssueLocationStore, IDisposable
+    {
+        void Register(ObservableCollection<IAnalysisIssueVisualization> issueVisualizations);
```

Comment:
I'm debating a bit if it should be `ReadOnlyObservableCollection`. It makes more sense and would protect the data from accidental changes, but it makes the usage much more complicated, as the callers have to call Unregister with the exact same wrapper instance that they used to call Register (which isn't a trivial assumption). I can add tests for the callers that verify that it's the same ref instance.
WDYT?

Reply:
Doesn't the caller have to provide the same instance anyway? Otherwise `ObservableIssueVisualizations.Remove(issueVisualizations)` won't be able to remove it, or to unregister from the event.

Reply:
You're right. I changed Register to
`IDisposable Register(ReadOnlyObservableCollection<IAnalysisIssueVisualization> issueVisualizations);`